internal/wallet/wallet.go

```diff
@@ -111,7 +112,7 @@ func Create(db walletdb.DB, params *chaincfg.Params, seed, pubPass,
 
 // Open loads a wallet from the passed db and public pass phrase.
 func Open(db walletdb.DB, pubPass []byte,
-	params *chaincfg.Params) (Wallet, error) {
+	params *chaincfg.Params) (*wallet, error) {
```

Comment:
I think this function should return a struct instead of an interface, because the wallet struct already fulfills the Wallet interface.

What do you think? @Jwata

Reply:
This [comment](https://github.com/dgarage/dlc/pull/45#discussion_r230271440
) addresses the issue

Reply:
As golint tells us, caller doesn't know about wallet because it's private.
> exported func Open returns unexported type *wallet.wallet, which can be annoying to use

Related to this discussion.
https://github.com/golang/lint/issues/210